### PR TITLE
feat(range-slider): use `rheostat` as slider component

### PR DESCRIPTION
**Pull request by iam4x**
See #2112

### CODE REVIEW

File: src/components/Slider/Slider.js [hunk 1]
```diff
@@ -1,92 +1,99 @@
...
+    refine(values);
   }
 
-  // we are only interested in rawValues
-  handleChange(formattedValues, handleId, rawValues) {
-    this.props.onChange(rawValues);
+  // creates an array number where to display a pit point on the slider
+  computeDefaultPitPoints({min, max}) {
+    const totalLenght = max - min;
+    const steps = 34;
```
Comment (bobylito) [inline src/components/Slider/Slider.js]: Why 34?

File: src/components/Slider/Slider.js [hunk 2]
```diff
@@ -1,92 +1,99 @@
...
+    const {refine} = this.props;
+    refine(values);
   }
 
-  // we are only interested in rawValues
-  handleChange(formattedValues, handleId, rawValues) {
-    this.props.onChange(rawValues);
+  // creates an array number where to display a pit point on the slider
+  computeDefaultPitPoints({min, max}) {
+    const totalLenght = max - min;
```
Comment (bobylito) [inline src/components/Slider/Slider.js]: should be `totalLength`

File: src/components/Slider/Slider.js [hunk 3]
```diff
@@ -1,92 +1,99 @@
...
+      // bug with `key={ 0 }` and preact, see https://github.com/developit/preact/issues/642
+      .map(pitPoint => pitPoint === 0 ? 0.00001 : pitPoint);
+
+    return pitPoints;
+  }
+
+  // creates an array of values where the slider should snap to
+  computeSnapPoints({min, max, step}) {
+    const r = range(min, max, step);
+    return r.includes(max) ? r : [...r.slice(0, -1), max];
```
Comment (bobylito) [inline src/components/Slider/Slider.js]: Range never includes max, https://lodash.com/docs/4.17.4#range Not sure what we're trying to achieve, but there might be an edge case if the last element of `r` is `max - step`.

Also mutating r would be more conservative. `r.slice(0, -1).push(max)` or even using `splice`.

File: src/components/Slider/Slider.js [hunk 4]
```diff
@@ -1,92 +1,99 @@
...
-    React.PropTypes.arrayOf(
-      React.PropTypes.shape({
-        to: React.PropTypes.func,
-      })
-    ),
-  ]),
-};
-
-export default autoHideContainerHOC(headerFooterHOC(RawSlider));
+export default autoHideContainerHOC(headerFooterHOC(Slider));
```
Comment (bobylito) [inline src/components/Slider/Slider.js]: 👍  A preview of some changes to come

File: src/widgets/range-slider/range-slider.js [hunk 5]
```diff
@@ -72,15 +70,11 @@ const usage = `Usage:
 rangeSlider({
   container,
   attributeName,
-  [ tooltips=true ],
-  [ templates.{header, footer} ],
```
Comment (bobylito) [inline src/widgets/range-slider/range-slider.js]: We probably don't want to remove the templates from the usage right? Nor the cssClasses, collapsible or autoHideContainer?

File: src/components/Slider/Slider.js (hunk 1 above)
Comment (iam4x) [inline src/components/Slider/Slider.js]: Backport of react-nouislider number of markers on the range

File: src/components/Slider/Slider.js (hunk 3 above)
Comment (iam4x) [inline src/components/Slider/Slider.js]: Didn't notice it will NEVER include it, we can have in the end:

```js
return [...range(min, max, step), max]
```

**Comment by redox** [comment]
That's awesome 👍

File: src/components/Slider/Slider.js (hunk 1 above)
Comment (bobylito) [inline src/components/Slider/Slider.js]: Mok :)

File: docgen/src/guides/migration.md [hunk 6]
```diff
@@ -88,6 +88,16 @@ As of now, we consider the engine used to build the widgets in InstantSearch.js
 as an implementation detail. Since we do not expose it anymore, we'll be able
 to change it and use the best solution for each release.
 
+## RangeSlider widget is using Rheostat as Slider Component
+
+Since we had unclosed issues with the old slider, we switched to a new
```
Comment (Haroenv) [inline docgen/src/guides/migration.md]: I would rephrase this more like `it had more features and we want to use it`, now it’s a bit negative

File: src/components/Slider/__tests__/Slider-test.js [hunk 7]
```diff
@@ -1,76 +1,23 @@
...
-          stepped: true,
-          values: [0, 50, 100],
-        } }
-        range={ {min: props.range.min, max: props.range.min + 0.0001} }
-        start={ props.start }
-        disabled
+describe('Slider', () => {
+  it('should render correctly', () => {
+    const tree = renderer.create(
+      <Slider
```
Comment (Haroenv) [inline src/components/Slider/__tests__/Slider-test.js]: Why do you explicitly render this, is this for more than shallow rendering?

File: src/components/Slider/Slider.js [hunk 8]
```diff
@@ -1,92 +1,99 @@
-import React from 'react';
-import omit from 'lodash/omit';
+import times from 'lodash/times';
+import range from 'lodash/range';
+import has from 'lodash/has';
 
-import Nouislider from 'react-nouislider';
+import React, {Component, PropTypes} from 'react';
```
Comment (iam4x) [inline src/components/Slider/Slider.js]: It's dealt properly with preact for now, we can have an issue open for v2.x 👍

File: docgen/src/guides/migration.md (hunk 6 above)
Comment (iam4x) [inline docgen/src/guides/migration.md]: It's negative for the past v1, so good for us nope?

File: src/components/Slider/__tests__/Slider-test.js (hunk 7 above)
Comment (iam4x) [inline src/components/Slider/__tests__/Slider-test.js]: Shallow rendering will only render the top component, so in that case it would have been the `<AutohideContainerHOC />`.

I just render it like this to prevent further regression with snapshot testing, it's a dummy test case. Look for `range-slider-test.js` on the widget itself where I test the rendering and the logic 👍

File: docgen/src/guides/migration.md (hunk 6 above)
Comment (Haroenv) [inline docgen/src/guides/migration.md]: I don’t agree, we can say that it’s better for the reasons that it’s better, but I don’t think we should say that it was bad before.
